File: app/src/main/java/com/john/kot/arch/recyclerview/RecycleViewModel.java
package com.john.kot.arch.recyclerview;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

public class RecycleViewModel extends ViewModel {

    private MutableLiveData<String[]> recycleLiveData;

    public LiveData<String[]> getRequestData() {
        if (recycleLiveData==null){
            recycleLiveData = new MutableLiveData<>();
        }
        return  recycleLiveData;
    }

    public void requestRecycleData(){
//        recycleLiveData.setValue();
    }


}

File: app/src/main/java/com/john/kot/arch/recyclerview/RecycleViewModel.kt
package com.john.kot.arch.recyclerview

import androidx.lifecycle.ViewModel

class RecycleViewModel : ViewModel() {

    /**
     * Generates Strings for RecyclerView's adapter. This data would usually come
     * from a local content provider or remote server.
     */
    fun initDataset(): List<String> {
        return arrayOf("50", "100", "150", "200", "Others").toList()
    }
}